quick_purchase/models/product_product.py

```diff
@@ -129,3 +129,16 @@ def search(self, args, offset=0, limit=None, order=None, count=False):
 
         return super(ProductProduct, self).search(
             args, offset=offset, limit=limit, order=order, count=count)
+
+    @api.multi
+    def button_return_purchase(self):
+        self.ensure_one()
+        purchase_id = self.env.context.get('purchase_id')
```

Comment:
if you test if purchase_id is defined, I think you should write .get('purchase_id', False).

Reply:
I don't think it's necessary.
http://www.tutorialspoint.com/python/dictionary_get.htm
The default value is perfect in my case.

Reply:
you're right ! thanks !